Decred Wallet/Features/History/TransactionHistoryViewController.swift

```diff
@@ -55,140 +60,135 @@ class TransactionHistoryViewController: UIViewController {
-            self.btnFilter.setTitle(LocalizedStrings.sent.appending(" (").appending(String(self.Filtercontent.count)).appending(")"), for: .normal)
-            self.tableView.reloadData()
-            break
-        case 2:
-            // TODO: Remove after next dcrlibwallet update
-            self.Filtercontent = self.mainContens.filter{$0.Direction == 1 && $0.Type == GlobalConstants.Strings.REGULAR}
-            self.btnFilter.setTitle(LocalizedStrings.received.appending(" (").appending(String(self.Filtercontent.count)).appending(")"), for: .normal)
-            self.tableView.reloadData()
-            break
-        case 3:
-            // TODO: Remove after next dcrlibwallet update
-            self.Filtercontent = self.mainContens.filter{$0.Direction == 2 && $0.Type == GlobalConstants.Strings.REGULAR}
-            self.btnFilter.setTitle(LocalizedStrings.yourself.appending(" (").appending(String(self.Filtercontent.count)).appending(")"), for: .normal)
-            self.tableView.reloadData()
-            break
-        case 4:
-            self.Filtercontent = self.mainContens.filter{$0.Type == GlobalConstants.Strings.REVOCATION || $0.Type == GlobalConstants.Strings.TICKET_PURCHASE || $0.Type == GlobalConstants.Strings.VOTE}
-            self.btnFilter.setTitle(LocalizedStrings.staking.appending(" (").appending(String(self.Filtercontent.count)).appending(")"), for: .normal)
-            self.tableView.reloadData()
-            break
-        case 5:
-            self.Filtercontent = self.mainContens.filter{$0.Type == GlobalConstants.Strings.COINBASE}
-            self.btnFilter.setTitle("Coinbase (".appending(String(self.Filtercontent.count)).appending(")"), for: .normal)
-            self.tableView.reloadData()
-            break
-        default:
-            self.Filtercontent = self.mainContens
-            self.btnFilter.setTitle(LocalizedStrings.all.appending(" (").appending(String(self.Filtercontent.count)).appending(")"), for: .normal)
-            self.tableView.reloadData()
-        }
-    }
-    
     func updateFilterDropdownItems() {
-        let sentCount = self.mainContens.filter{$0.Direction == 0}.count
-        let ReceiveCount = self.mainContens.filter{$0.Direction == 1}.count
-        let yourselfCount = self.mainContens.filter{$0.Direction == 2}.count
-        let stakeCount = self.mainContens.filter{$0.Type.lowercased() != "Regular".lowercased()}.count
-        let coinbaseCount = self.mainContens.filter{$0.Type == GlobalConstants.Strings.COINBASE}.count
+        let sentCount = self.allTransactions.filter{$0.Direction == DcrlibwalletTxDirectionSent}.count
+        let receiveCount = self.allTransactions.filter{$0.Direction == DcrlibwalletTxDirectionReceived}.count
+        let yourselfCount = self.allTransactions.filter{$0.Direction == DcrlibwalletTxDirectionTransferred}.count
+        let stakeCount = self.allTransactions.filter{$0.Type.lowercased() != "Regular".lowercased()}.count
```

Comment:
Instead of using hardcoded "Regular", use DcrlibwalletTxTypeRegular. Also, no need to lowercase `$0.Type` and `DcrlibwalletTxTypeRegular`

Reply:
fixed in 28f46d1